src/config.ts

```diff
@@ -713,7 +713,40 @@ export default {
    *
    * @type {Array.<Number>}
    */
-  SEGMENT_PRIORITIES_STEPS : [6, 14],
+  SEGMENT_PRIORITIES_STEPS : [ 2,   // Step 0: 0-2
+                               4,   // Step 1: 2-4 => last high priority
+                               8,   // Step 2: 4-8
+                               12,  // Step 3: 8-12 => first cancellable priority
+                               18,  // Step 4: 12-18
+                               25], // Step 5 18-25
+                                    // Step 6: 25+
+
+  /**
+   * Some segment requests are said to be "high priority".
+   *
+   * Requests in that category once done will cancel any segment request that
+   * has a low priority step (see `SEGMENT_PRIORITIES_STEPS`) - meaning a
+   * priority step equal to `MIN_CANCELABLE_PRIORITY` or more.
+   *
+   * Enter here the last priority step that is considered high priority
+   * (beginning by the step `0`).
+   * @type {number}
+   */
+  MAX_HIGH_PRIORITY_LEVEL: 1, // priority step 1 is the last "high priority"
+
+  /**
+   * Enter here the first priority step (see `SEGMENT_PRIORITIES_STEPS`) that
+   * will be considered as low priority.
+   *
+   * Segment requests with a low priority will be cancelled if a high priority
+   * segment request (see MAX_HIGH_PRIORITY_LEVEL) is scheduled while they are
+   * pending.
+   *
+   * This number should be strictly superior to the value indicated in
+   * `MAX_HIGH_PRIORITY_LEVEL`.
+   * @type {number}
+   */
+  MIN_CANCELABLE_PRIORITY: 3, // priority step 3 onward can be cancelled
```

Comment:
Why is it a step and not a priority ? It seems more logical to me that this attributes means something in terms of priority.

Reply:
? I wonder if there's some confusion between what we consider as steps and priorities.

For me, both means the same thing, perhaps the documentation is not too good with that.

Basically, priority raise step by step depending on the distance from the current position.
For example with a `SEGMENT_PRIORITIES_STEPS` set at `[4, 7]`, we will have 3 priority numbers/steps:
  - the priority number/step `0` for segments that begin in <4 seconds
  - the priority number/step `1` for segments beginning at a time `t` where `4 <= t < 7` (in seconds)
  - the priority number/step `2` for segments beginning in more than 7 seconds
(a lower priority number/step means a higher priority)

Here I just conflated the notion of a priority number with the term "priority step", which are both term I use interchangeably.
 - "Priority number" because the priority of a segment is in reality a number
 - "Priority step" because it evolves step by step, not continuously.

What may seem weird is that I also 0-indexed the notion of a priority step, where it is not really natural to think about a 0th step.

Reply:
I tried updating the documentation to make it more clear.

Also now, the "step" word is just used to explain the notion of having a stepped algorithm for scheduling segment requests. It is now 1-indexed in the documentation (because it is only a "human" concept) and directly translated to a corresponding "priority number" which is explained as the priority value it will have in the code. The priority number is still 0-indexed.